crates/db_queries/src/source/post.rs

```diff
@@ -260,6 +260,20 @@ impl Readable<PostReadForm> for PostRead {
   }
 }
 
+impl DeleteableOrRemoveable for Post {
+  fn blank_out_deleted_or_removed_info(mut self) -> Self {
+    self.name = "".into();
+    self.url = None;
+    self.body = None;
+    self.embed_title = None;
+    self.embed_description = None;
+    self.embed_html = None;
+    self.thumbnail_url = None;
+
+    self
+  }
+}
+
```

Comment:
This is the main thing, adding a trait for these types that allows you to blank out some fields.